mono/utils/mono-time.c

```diff
@@ -161,6 +178,31 @@ mono_100ns_ticks (void)
 }
 
 /*
+ * Returns user + system time for a thread in 100ns units.
+ */
+gint64
+mono_thread_cpu_time (void)
+{
+#if defined(__linux__)
+	struct timespec ts;
+	clock_gettime (CLOCK_THREAD_CPUTIME_ID, &ts);
+	return ts.tv_sec * 10000000 + ts.tv_nsec / 100;
+#elif defined(PLATFORM_MACOSX)
+	mach_msg_type_number_t count = THREAD_BASIC_INFO_COUNT;
+	thread_basic_info_data_t info;
+
+	if (thread_info (pthread_mach_thread_np (pthread_self ()), THREAD_BASIC_INFO,
```

Comment:
This is the code to use the Mach API: https://github.com/schani/mono/commit/3d5fb7bb0aa07ec71ef9bf412be1c126899f596e

The include of `pthread.h` in there is erroneous, I think.  It should only be necessary for the PThread API.